applications/rag/README.md

```diff
@@ -1,192 +1,210 @@
-    1. Go to https://console.cloud.google.com/net-services/dns/zones, select the zone and click ADD STANDARD, fill in your domain name and public IP address.
-    2. Run command: `gcloud dns record-sets create <domain address>. --zone=<zone name> --type="A" --ttl=<ttl in seconds> --rrdatas="<public ip address>"`
+This section generates the vector embeddings for your input dataset. Currently, the default dataset is [Netflix shows](https://www.kaggle.com/datasets/shivamb/netflix-shows). We will use a Jupyter notebook to run a Ray job that generates the embeddings & populates them into the `pgvector` instance created above.
 
-6. Optionally, enable Cloud Data Loss Prevention (DLP)
+Set env vars (change to the namespace in `workloads.tfvars` if using a custom `kubernetes_namespace`):
 
-    We have two ways to enable the api:
+```
+export NAMESPACE=rag
+```
 
-    1. Go to https://console.developers.google.com/apis/api/dlp.googleapis.com/overview click enable api.
-    2. Run command: `gcloud services enable dlp.googleapis.com`
+1. Connect and login to JupyterHub:
+   * If IAP is disabled (`jupyter_add_auth = false`):
+        - Port forward to the JupyterHub service: `kubectl port-forward service/proxy-public -n ${NAMESPACE} 8081:80 &`
+        - Go to `localhost:8081` in a browser
+        - Login with these credentials:
+            * username: admin
+            * password: use `terraform output jupyterhub_password` to fetch the password value
+   * If IAP is enabled (`jupyter_add_auth = true`):
+        - Fetch the domain: `terraform output jupyterhub_uri`
+        - If you used a custom domain, ensure you configured your DNS as described above. This can be skipped if using `nip.io`.
+        - Verify the domain status is `Active`:
+            - `kubectl get managedcertificates jupyter-managed-cert -n ${NAMESPACE} --output jsonpath='{.status.domainStatus[0].status}'`
+            - Note: This can take up to 20 minutes to propagate.
+        - Once the domain status is Active, go to the domain in a browser and login with your Google credentials.
+        - To add additional users to your JupyterHub application, go to [Google Cloud Platform IAP](https://console.cloud.google.com/security/iap), select the `rag/proxy-public` service and add principals with the role `IAP-secured Web App User`.
+
+2. Load the notebook:
+    - Once logged in to JupyterHub, choose the `CPU` preset with `Default` storage. 
+    - Go to File -> Open From URL & upload the notebook `rag-kaggle-ray-sql.ipynb` from `https://raw.githubusercontent.com/GoogleCloudPlatform/ai-on-gke/main/applications/rag/example_notebooks/rag-kaggle-ray-sql-latest.ipynb`. This path can also be found by going to the [notebook location](https://github.com/GoogleCloudPlatform/ai-on-gke/blob/main/applications/rag/example_notebooks/rag-kaggle-ray-sql-latest.ipynb) and selecting `Raw`.
```

Comment:
Let's remove the last sentence ("This path can also be found..."). Redundant info tends to confuse.